### ISSUE TITLE
Nagivation doesnt work with static meshes

### ISSUE THREAD

**Issue by YerkoAndrei** [stride3d/stride#2326]
**Release Type**: Official Release

**Version**: 4.2.0.2149

**Platform(s)**: Windows

**Describe the bug**
Nagivation mesh ignore all static meshes

**To Reproduce**
1. Create new project
2. Add simple navigation configuration as in the official tutorial
3. Add a plane and some cube
4. Add static colliders for both
5. Add infinite mesh to plane
6. On collider shapes of the cube, add static mesh and reference itself

**Expected behavior**
Navigation mesh should update and use the cube as a wall or obstacle

**Screenshots**
With static mesh
![1](https://github.com/stride3d/stride/assets/19843418/c0ac7d86-9b81-48ea-84c4-c6f5f86000fd)

With cube shape (how it should look)
![2](https://github.com/stride3d/stride/assets/19843418/fa22ca03-0f2a-418d-a3f4-b9a243e8b20c)

**Additional context**
* Its not a visual problem, characters navigation doesnt work
* Using a convex hull works good, but it cant be used as a workaround
* Creating a collider shape asset doesnt work
* I tried on master and it doesnt work either

**Comment by Doprez** [comment]
If you enlarge the box does the mesh still work with the surface of the mesh? Im curious because I know I used a static mesh at one point for terrain on the old recast nav mesh and it worked for that at least.

**Comment by YerkoAndrei** [comment]
Nope, i cant make it work, i have tried with fbx and glb, so its not that.

Im looking the code and the ColliderShape of the StaticCollider is alway null when using a StaticMesh. I think thats the problem.
When the PhysicsComponent try to ComposeShape, it call CreateShape with a null service, wich is only used by Asset and StaticMesh (just these 2 fails).
I dont undestand yet what is an EntityManager and why is null, so if someone know someting related please help.

**Comment by Doprez** [comment]
I should be able to look into this a bit this week. The file type should not matter so long as the collider is created, it uses the [information from Bullet to create the mesh](https://github.com/stride3d/stride/blob/dbb125e19a48715dd9fa7cf200a7d3f75f2fbb78/sources/engine/Stride.Navigation/NavigationMeshBuilder.cs#L569).

Can you also try to downgrade your project to 4.2.0.2122? Im wondering if one of the PRs may have had a larger affect than intended. https://github.com/stride3d/stride/pull/2262

**Comment by YerkoAndrei** [comment]
tried 4.2.0.2188, 4.2.0.2149, 4.2.0.2122 and 4.2.0.2042, not working on any

**Comment by Doprez** [comment]
got it working in my branch https://github.com/Doprez/stride/tree/fix-static-collider-mesh

seems like it has to do with reading asset data that isn't immediately available in editor for some reason.

![image](https://github.com/stride3d/stride/assets/73259914/03532642-3c2c-4d2b-84c7-ccb4b24d6263)

**Comment by Doprez** [comment]
I did a quick test to make sure it wasn't just working for procedural meshes:
![image](https://github.com/stride3d/stride/assets/73259914/21f920c3-ff79-4391-b107-33829962bd18)

The only quirk that I can see now is you have to move the physics collider to update the debug mesh the first time.

**Comment by YerkoAndrei** [comment]
its not working on my machine, i tried that 2 weeks ago and it didnt work :(
ill try a project from zero

**Comment by Doprez** [comment]
I did some more messing around and a new issue now is the editor full crashes when more than one static mesh is added to the collider shapes.

**Comment by Doprez** [comment]
So, something thats odd but seems to work is if you create your physics meshes while the navigation stuff is deleted it seems to be fine. 

the crash seems to be the fault of `NavigationMeshBuilder` line 479 when its expecting an updated component that may be out of date or too early? looking into that now.

**Comment by Doprez** [comment]
Wait.... even in 4.2.0.2188 I can get the static meshes to work. As long as I don't use a procedural mesh at least.

![image](https://github.com/stride3d/stride/assets/73259914/c2b78a2d-36d9-4aef-a17b-19e625c72dd6)

The current issues seem to be related to using procedural meshes as the static mesh or trying to create a compound mesh for the static mesh.

Another important thing to note is that there is a bit of lag with how meshes are updated due to transforms being updated after the mesh is built https://github.com/stride3d/stride/blob/904371f6c62e7aec870d12e85dd881f38f5c15c1/sources/engine/Stride.Navigation/DynamicNavigationMeshSystem.cs#L134
